crates/viewer/re_view_spatial/src/contexts/transform_context.rs

```diff
@@ -550,17 +551,36 @@ fn query_and_resolve_tree_transform_at_entity(
     if let Some(translation) = result.component_instance::<Translation3D>(0) {
         transform = glam::Affine3A::from(translation);
     }
-    if let Some(rotation_quat) = result.component_instance::<RotationAxisAngle>(0) {
-        transform *= glam::Affine3A::from(rotation_quat);
+    if let Some(axis_angle) = result.component_instance::<RotationAxisAngle>(0) {
+        if let Ok(axis_angle) = glam::Affine3A::try_from(axis_angle) {
+            transform *= axis_angle;
+        } else {
+            // Invalid transform.
+            return None;
+        }
     }
-    if let Some(rotation_axis_angle) = result.component_instance::<RotationQuat>(0) {
-        transform *= glam::Affine3A::from(rotation_axis_angle);
+    if let Some(quaternion) = result.component_instance::<RotationQuat>(0) {
+        if let Ok(quaternion) = glam::Affine3A::try_from(quaternion) {
+            transform *= quaternion;
+        } else {
+            // Invalid transform.
+            return None;
+        }
     }
     if let Some(scale) = result.component_instance::<Scale3D>(0) {
+        if scale.x() == 0.0 || scale.y() == 0.0 || scale.z() == 0.0 {
```

Comment:
```suggestion
        if scale.x() == 0.0 && scale.y() == 0.0 && scale.z() == 0.0 {
```
I believe it should be valid to collapse a single dimension onto a plane via scaling.

Reply:
good point, for some things it is! 👍